### ISSUE TITLE
Request: Error for ${} around a file path with multiple variables

### ISSUE THREAD

**Issue by StandingPadAnimations** [koalaman/shellcheck#2786]
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

In Bash, having `${}` around multiple variables and an expression (like `version/dir`) can cause massive consequences, as mentioned here (tldr: every Blender addon was deleted for every version of Blender): https://github.com/TheDuckCow/MCprep/pull/422#issuecomment-1606525786

#### Here's a snippet or screenshot that shows the problem:

```sh

rm -rf "${i/$NAME:?}/"

```

#### Here's what shellcheck currently says:
Nothing. Ironically the example line was to fix another shellcheck error related to not checking if a variable is null

#### Here's what I wanted or expected to see:
An error saying something like "`${}` should be around each individual variable"

**Comment by StandingPadAnimations** [comment]
> How was it intended to be expanded, though?

Where `i` and `NAME` are expanded with a `/` in the middle. Given `i=12345; NAME=45; echo "${i/NAME:?}/"`, I intended for the result to be `12345/45` (quite dumb I know) 


> From `man bash`, Parameter Expansions on ${parameter/pattern/string}, first paragraph: "If string is null, matches of pattern are deleted and the / following pattern may be omitted."

Yeah that is far from what I intended

**Comment by wileyhy** [comment]
Nah, parameter expansions are tricky.

"${i}/${NAME}"

...is what you were looking for, then. Alternatively, you could use...

"${i:?}/${NAME:?}"/

...with a trailing backslash and parameter expansions that will halt the
script if a variable is unset or null. ...however in POSIX mode, the shell
will not exit. A custom error message can be included at 'word' with this
syntax:

"${parameter:?word}"
